plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/reflection/HtmlReflection.java

```diff
@@ -320,21 +248,83 @@ public static HtmlElement findBlock(Page page, String blockPath) throws NoSuchEl
     }
 
     /**
-     * See {@link HtmlReflectionUtils#findBlocks(String, Object, boolean)} for description.
+     * See {@link HtmlReflection#findBlocks(String, Object, boolean)} for description.
      * Search is being performed on a current page
      *
      * @param page the page on which the method will be executed
      * @param blockPath full path or just a name of the block to search
      * @return list of objects that were found by specified path
      */
-    public static List<HtmlElement> findBlocks(Page page, String blockPath) throws NoSuchElementException {
+    public List<HtmlElement> findBlocks(Page page, String blockPath) throws NoSuchElementException {
         try {
             return findBlocks(blockPath, page, false);
         } catch (IllegalAccessException ex) {
             throw new FactoryRuntimeException(String.format("Internal error during attempt to find a block '%s'", blockPath), ex);
         }
     }
 
+    /**
+     * Finds blocks by required path/name in the given context. Block is a
+     * class that extends HtmlElement. If blockPath contains delimiters, it
+     * will be treated as a full path, and block should be located by the
+     * exactly that path. Otherwise, recursive search via all blocks is
+     * being performed
+     *
+     * @param blockPath full path or just a name of the block to search
+     * @param context object where the search will be performed
+     * @param returnFirstFound whether the search should be stopped on a
+     * first found block (for faster searches)
+     * @return list of found blocks. could be empty
+     * @throws IllegalAccessException if called with invalid context
+     */
+    private List<HtmlElement> findBlocks(String blockPath, Object context, boolean returnFirstFound)
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. ([squid:S3776](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS3776))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=101&issues=AWbDgFYRLpqqSUlctfyF&open=AWbDgFYRLpqqSUlctfyF)

[ISSUE_KEY]: AWbDgFYRLpqqSUlctfyF